.buildkite/pipeline.yml

```diff
@@ -25,5 +25,21 @@ steps:
 
   - label: ":golang-lint: Lint"
     key: lint
-    command: make lint
+    command: rm -rf .git && make lint
```

Comment:
I couldn't find an equivalent of `-buildvcs=false` the running `golangci-lint` and I would repeatedly get the error:
```console
main.go:1: : error obtaining VCS status: exit status 128
Use -buildvcs=false to disable VCS stamping. (typecheck)
```

I couldn't find a better (and simple) workaround and I think `rm -rf .git` should be ok to do on CI, but I'd be happy if someone has a suggestion!

Reply:
Did you try https://github.com/golangci/golangci-lint/issues/4033#issuecomment-1686025933 ?

Reply:
I've seen this post while working on #2, will try now!

Reply:
Also:
 - Even if this would be low impact to `rm -rf .git` on CI because this runs on a throwaway clone of the repo, the fact that this `rm` is necessary to make `make lint` pass means that running `make lint` locally from your Mac would fail, right? While the goal especially with our setups on our tooling repos to use a `Makefile` is so that we can ideally run the same commands on CI and locally for debugging.
 - And in fact I'm confused, because I just tried running `docker run --rm -it -v ./:/app -w /app golangci/golangci-lint:v1.60.1 /bin/bash -c "go mod vendor && golangci-lint run -v --modules-download-mode vendor --timeout=10m"` from my Mac and it worked without any complaint about `.git` 🤔

Reply:
> so that we can ideally run the same commands on CI and locally for debugging.

Agreed. 

>  I just tried running [...] from my Mac and it worked 

Same with me -- it works locally and [fails on CI](https://buildkite.com/automattic/hostmgr-packer-plugin/builds/4#01915b35-3e9b-4ed6-8f53-fb07f8b73c89/123-237), I guess due to permissions? I will try what folks used on that thread.